cf-agent/verify_files_utils.c

```diff
@@ -3694,6 +3694,44 @@ bool CfCreateFile(EvalContext *ctx, char *file, const Promise *pp, Attributes at
             return false;
         }
     }
+    else if (attr.file_type && !strncmp(attr.file_type, "fifo", 5))
+    {
+        if (!DONTDO)
+        {
+            mode_t saveumask = umask(0);
+            mode_t filemode = 0600;
+
+            if (PromiseGetConstraintAsRval(pp, "mode", RVAL_TYPE_SCALAR) == NULL)
+            {
+                /* Relying on umask is risky */
+                filemode = 0600;
+                Log(LOG_LEVEL_VERBOSE, "No mode was set, choose plain file default %04jo", (uintmax_t)filemode);
+            }
+            else
+            {
+                filemode = attr.perms.plus & ~(attr.perms.minus);
+            }
+
+            MakeParentDirectory(file, attr.move_obstructions);
+
+            char errormsg[CF_BUFSIZE];
+            if (!mkfifo(file, filemode))
+            {
+                snprintf(errormsg, sizeof(errormsg), "(mkfifo: %s)", GetErrorStr());
+                umask(saveumask);
+                return false;
+            }
+
+            umask(saveumask);
+            return true;
+        }
+        else
+        {
+            cfPS(ctx, LOG_LEVEL_ERR, PROMISE_RESULT_WARN, pp, attr, "Warning promised, need to create fifo '%s'", file);
+            *result = PromiseResultUpdate(*result, PROMISE_RESULT_WARN);
+            return false;
+        }
```

Comment:
Can you put a ``#ifndef _WIN32`` around the whole block (starting from the ``DONTDO`` statement), and print a warning that the operation is unsupported on Windows?